tools/vulnerability-scan/wss-scan.sh

```diff
@@ -47,33 +49,67 @@ basepath=$baseDirPath"/repos"
 
 echo "Cleaning up scan directories if already present"
 rm -rf $basepath
+echo "Cleaning up temp file that will affect scan"
+rm -rf /tmp/ws*
 
 mkdir -p $basepath
  
 # clone the desired Repos for scanning 
 for repo in $gitRepos
 do
   echo "Cloning repo "$gitBasePath$repo
-  git clone "$gitBasePath$repo".git $basepath"/"$repo
+  if [ ${repo} == "OpenSearch" ]
+  then
+    echo "Cloning "$repo" at branch 1.x"
+    git clone -b 1.x "$gitBasePath$repo".git ${basepath}/${repo}
+    # /qa/ in OpenSearch repo contains all files related to the previous version of ODFE.
+    # WhiteSource will attept to download them and take hours to build or timeout error. 
+    # Remove /qa/ because it's irrelevant to OpenSearch. 
+    rm -rf ${basepath}/${repo}/qa
+  else
+    echo "Cloning "$repo" at main branch" 
+    git clone "$gitBasePath$repo".git ${basepath}/${repo}
+  fi
 done
 
 echo -n > info.txt
 target='OpenSearch'
-mvn -v; node -v; npm -v; yarn -v
 
 # scan the Repos using the WhiteSource Unified Agent
 for repo in $gitRepos
 do
   repo_path=$basepath"/"$repo
   if [ -d "$repo_path" ]
   then
+    cd $repo_path
```

Comment:
Let's use absolute path's everywhere instead of `cd` in a folder. It would make it very easy then see and keep track of the files we are referring to. It would otherwise be difficult to keep track and extend this script later.

Reply:
After testing, going into the repository folder and running the scan inside it will improve the scan stability with WhiteSource. I changed all with absolute path though to make them more clear.

Reply:
We can remove the cd from here then?